stock_request_purchase/models/stock_request.py

```diff
@@ -38,6 +39,14 @@ def _check_purchase_company_constrains(self):
                 )
             )
 
+    def action_cancel(self):
+        """Propagate the cancellation to the generated purchase orders."""
+        res = super().action_cancel()
+        self.sudo().purchase_ids.filtered(
```

Comment:
What happens if several products from other stock request/MTO/etc are in the same purchase order? I think you should check that the PO is only from this stock request.